web-ui/src/main/resources/catalog/components/utility/UtilityService.js

```diff
@@ -350,11 +350,25 @@
    */
   module.filter('gnLocalized', function() {
     return function(obj, lang) {
-      if (angular.isObject(obj)) {
-        return obj[lang] ? obj[lang] : (obj[Object.keys(obj)[0]] || '');
+      if (angular.isObject(obj)) {   
+        if (lang&&obj[lang]){ 
+            return obj[lang];
+        } else {
+          //in some cases the object has a key: 'type', which should not be returned as the first language
```

Comment:
In which case do we have this 'type' key ?

Reply:
when adding a service layer i noticed the url had 
`{"type":"onlineresource","dut":"http://..."}`